### PR TITLE
Manifest uuid

### PR DESCRIPTION
**Pull request by pkarman**
Trello: https://trello.com/c/dA6hnR2L

UUID is created by postgresql on insert, following v4 UUID spec: http://blog.arkency.com/2014/10/how-to-start-using-uuid-in-activerecord-with-postgresql/

### CODE REVIEW

File: app/models/manifest.rb [hunk 1]
```diff
@@ -43,4 +43,15 @@ def self.authorized_search(params, current_user=nil)
     dsl = QueryDSL.new(params: params, current_user: current_user)
     search(dsl)
   end
+
+  def as_public_json(options={})
```
Comment (jessieay) [inline app/models/manifest.rb]: I think that with our use of `ManifestSerializer` we shouldn't need this - we can just defined `id` on the serializers as `object.uuid`

File: db/migrate/20160112190459_add_uuid_to_manifests.rb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+class AddUuidToManifests < ActiveRecord::Migration
+  def up
+    enable_extension 'uuid-ossp'
+    add_column :manifests, :uuid, :uuid
+
+    Manifest.all.each do |manifest|
```
Comment (jessieay) [inline db/migrate/20160112190459_add_uuid_to_manifests.rb:9]: should we throw a quick class definition of `Manifest` in here for forward compatibility? http://railsguides.net/change-data-in-migrations-like-a-boss/

File: app/controllers/api/v0/manifests_controller.rb [hunk 3]
```diff
@@ -25,26 +25,18 @@ def create
...
-        manifest = Manifest.find_by!("content -> 'generator' ->> 'manifest_tracking_number' = ?", params[:tracking_number])
-      end
+      manifest = find_manifest
     rescue ActiveRecord::RecordNotFound => _error
       render json: {}, status: 404
       return
     end
 
-    render json: manifest.to_json
+    render json: manifest.to_public_json
```
Comment (jessieay) [inline app/controllers/api/v0/manifests_controller.rb]: I think we should be using `ManifestSerializer.new(manifest).to_json` here

Comment (jessieay) [inline app/controllers/api/v0/manifests_controller.rb]: I had that in the PR that was closed, but the serializer did make it into the Rails PR :)

File: app/controllers/api/v0/manifests_controller.rb [hunk 4]
```diff
@@ -66,4 +58,20 @@ def manifest_params
       :manifest_tracking_number,
     )
   end
+
+  def find_manifest
+    if params[:id] || params[:uuid]
+      find_manifest_by_id
+    elsif params[:tracking_number]
+      find_manifest_by_tracking_number
```
Comment (jessieay) [inline app/controllers/api/v0/manifests_controller.rb:66]: niiice

**Comment by jessieay** [comment]
a couple of comments about how we're rendering JSON for manifest - let me know what you think!

**Comment by pkarman** [comment]
This is going to fail an Elasticsearch test until we merge #48

**Comment by pkarman** [comment]
I enabled ES server in 0189994 just to prove that tests will pass.

File: spec/serializers/manifest_serializer_spec.rb [hunk 5]
```diff
@@ -0,0 +1,10 @@
+require 'rails_helper'
+
+describe ManifestSerializer do
```
Comment (jessieay) [inline spec/serializers/manifest_serializer_spec.rb:3]: nice! I am never sure how or when to unit test serializers, but this seems like a great use case

**Comment by jessieay** [comment]
looks superb -- merging!

**Pull request merged**